pandas/sparse/array.py

```diff
@@ -59,7 +59,12 @@ def wrapper(self, other):
 
 
 def _sparse_array_op(left, right, op, name):
-    if np.isnan(left.fill_value):
+    if (np.isnan(left.fill_value) and np.isnan(right.fill_value) and
```

Comment:
Yes removing ``nanop`` simplifies the logic. Because data is sparse, it shouldn't affect to performance in most cases.